### PR TITLE
SW-5880 Added observation aggregates to subzones

### PR DESCRIPTION
**Pull request by tommylau523**
Implemented logic to fetch observation summary. Some changes as follow:
1) Added subzone stats in an observation by aggregating monitoring plot
2) Method to fetch latest observation per subzone
3) Methods to build up observation summary at zone/site level from the bottoms up. 
4) Added test cases for summary calculations: https://docs.google.com/spreadsheets/d/17-9VlElRgnXU7rnCPsesw6bM5c1Du7Z_ZIvSq8yJPbI/edit?usp=sharing (blue highlighted rows under plot stats reference an older observation)
5) Added observations CSV as a way to bulk import plants within an observation

Future work:
- Subzone stats test will be added in a follow-up PR
- Per species logic for zone/site will be implemented in a follow-up PR. It requires a method to combine monitoring plot species.

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2538** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2538?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/main/kotlin/com/terraformation/backend/tracking/api/ObservationsController.kt
Comment (tommylau523) [inline src/main/kotlin/com/terraformation/backend/tracking/api/ObservationsController.kt]: Not sure if we need this endpoint yet

File: src/main/kotlin/com/terraformation/backend/tracking/db/ObservationResultsStore.kt
Comment (tommylau523) [inline src/main/kotlin/com/terraformation/backend/tracking/db/ObservationResultsStore.kt]: On some more thought, I will build out more complete aggregate models and combine observation results and stats from the bottoms up.

**Comment by sgrimm** [comment]
I'm assuming I should hold off reviewing this for now, based on your last comment? Sounded like you were going to rework some of this.

File: src/main/kotlin/com/terraformation/backend/tracking/model/ObservationResultsModel.kt
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/tracking/model/ObservationResultsModel.kt:125]: Why do we need this interface? Doesn't look like anything is taking it as a parameter or using it as a type.

Maybe it's to avoid repeating the KDoc comments for similar fields? In that case, the comments here should be updated to not refer specifically to zones any more.

**Review by sgrimm — CHANGES_REQUESTED** [review]
This could really use some test coverage; it introduces new functionality and changes how some of the existing functionality works. This now supports partial observations, but there's no validation that it comes up with the correct statistics, and no validation that the subzone statistics are correct.

I think it should be possible to modify `ObservationResultsStoreTest` to test scenarios where only certain subzones are included in an observation, maybe using the spreadsheet from the Jira task as input like some of the existing test cases do.

**Comment by sgrimm** [comment]
I'm working on reviewing this one; probably won't finish today but I have some changes I'll want to push to this PR's branch before it's ready to merge.

File: src/main/kotlin/com/terraformation/backend/tracking/model/ObservationResultsModel.kt
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/tracking/model/ObservationResultsModel.kt]: Changing this to two fields for earliest and latest both because it's clearer and because everywhere this is used, the calling code unpacks it into two separate variables anyway.

**Review by sgrimm — APPROVED** [review]
Had some minor feedback on the previous version of this, but it looked good in general; I applied my suggested edits directly since Tommy is out.

**Pull request merged**
